### ISSUE TITLE
[Feature] Add emoji search to mobile, plug the gap on cross-platform inconsistency

### ISSUE THREAD

**Issue by JmillsExpensify** [Expensify/App#14098]
# Problem
Currently you can search for a specific emoji in the picker on web/desktop, but not in the mobile apps.

# Why is this important?
We expect mobile and desktop to have perfect parity, but this is not currently the case.

# Solution
Update the mobile emoji picker to also have a search feature
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018350435dc437a500</li>
        <li>Upwork Job ID: 1611497960351657984</li>
        <li>Last Price Increase: 2023-01-06</li>
    </ul>
</details>

**Comment by JmillsExpensify** [comment]
We're pulling this issue into the public /App repo, so I'm going to copy over some of the more important comments.

First up from @stitesExpensify 
> For posterity so that I remember when coming back to this:
> We use a bottom-docked popover for the emoji picker. This means that we can't use a keyboard avoiding view to move the picker above the keyboard and end up with something like this:
<img width="452" alt="2022-11-30_08-58-47" src="https://user-images.githubusercontent.com/42391420/204846718-e46e5c21-d357-4184-8495-f09f710983f4.png">

> I'm thinking that the solution is going to be either to change the type of modal we use, or somehow detect the keyboard being opened and manually moving the anchor point (which is currently not used for mobile).

**Comment by puneetlath** [comment]
I think we should just do this like Slack where the modal content gets pushed up along with the keyboard. 


https://user-images.githubusercontent.com/1371996/211122797-db4ae172-5c8f-4c24-90a7-4b7db7e73b1c.MP4

**Comment by shawnborton** [comment]
Yup, I agree with you @puneetlath

**Comment by JmillsExpensify** [comment]
Yeah, agreed. Slack does a nice job there.

**Comment by stitesExpensify** [comment]
Agreed!

**Comment by JmillsExpensify** [comment]
Btw, we need to get Margelo to comment on this issue so that we can assign them.

**Comment by JmillsExpensify** [comment]
Sounds like @hannojg might take this one over. Is that right?

**Comment by JmillsExpensify** [comment]
Updating this issue since my last comment, as Margelo as focused more on reactions at the moment. @stitesExpensify I've had several engineers reach out asking for something helpful to get involved with. Maybe we put this issue up for grabs in engineering-chat? Did you want to do the honors?

**Comment by JmillsExpensify** [comment]
Actually scratch that! We have a new Margelo member helping out now. @perunt shall we assign you to this issue as well?

**Comment by perunt** [comment]
I guess yes. I would be happy to take on this issue

**Comment by JmillsExpensify** [comment]
Cool, sound good! For clarity, the auto-suggestions for emoji short codes is more valuable in any case. We'll assign you for now, but if someone from our team has a free cycle and would like to pitch it up, we can also go that route as well.

**Comment by stitesExpensify** [comment]
Not overdue. I'm going to move this to weekly since @perunt is focused on auto-suggestions for now

**Comment by stitesExpensify** [comment]
Same update as last week

**Comment by perunt** [comment]
@stitesExpensify assembled issue, going to record video how it works now

**Comment by perunt** [comment]
https://user-images.githubusercontent.com/48593211/215893849-03cefed2-392b-464a-ae1a-bee602b8abb6.mp4

@stitesExpensify the rest videos are in the `Screenshots/Videos` section in the PR

**Comment by stitesExpensify** [comment]
Thanks for that, looking good! Are we auto-focusing the search bar which is causing the keyboard to pop up? I think the intended functionality would be for the emojis to be at the bottom of the screen for easy reachability, and then if the user taps the search bar we would move everything up for the keyboard and they can search

**Comment by puneetlath** [comment]
Is this full page modal a design pattern we currently have? I was assuming it'd be a bottom docked modal that slides up over the chat.

**Comment by perunt** [comment]
@stitesExpensify yes, I completely agree with your suggestion regarding the auto-focus of the search bar. It eliminates the flaky user experience and makes it more intuitive. I've attached a video without the auto-focus. Let me know if you'd like me to make any changes.

https://user-images.githubusercontent.com/48593211/216161108-2469bb9f-3d30-40e1-8cbd-ab93ac50612e.mp4

**Comment by stitesExpensify** [comment]
That looks great to me!

**Comment by stitesExpensify** [comment]
I did want to let you know that one of us is definitely going to run into a bunch of merge conflicts though 😅 https://github.com/Expensify/App/pull/14532

**Comment by puneetlath** [comment]
>However, you can see how the modal will look on larger screens in the video I attached to my previous response. It provides a better representation of the bottom modal

Ah cool. Yeah agree this looks great!

**Comment by JmillsExpensify** [comment]
Wow yeah, great work on this.

**Comment by stitesExpensify** [comment]
Great! Let me know when you're ready for me to start reviewing @perunt!

**Comment by MelvinBot** [comment]
This issue has not been updated in over ***15*** days. @JmillsExpensify, @stitesExpensify, @perunt eroding to ***Monthly*** issue.

P.S. Is everyone reading this sure this is really a near-term priority? Be brave: if you disagree, go ahead and close it out. If someone disagrees, they'll reopen it, and if they don't: one less thing to do!

**Comment by JmillsExpensify** [comment]
Coming back from OOO. What's the latest on this one?

**Comment by stitesExpensify** [comment]
AFAIK @perunt is still focused on emoji autosuggestion. Can you confirm please, Taras?

**Comment by JmillsExpensify** [comment]
Cool, @perunt will you be in Curacao with us? It'd be good to catch up.

**Comment by MelvinBot** [comment]
⚠️ Looks like this issue was linked to a **Deploy Blocker** [here](https://github.com/Expensify/App/issues/16636)

If you are the assigned CME please investigate whether the linked PR caused a regression and leave a comment with the results.

If a regression has occurred and you are the assigned CM follow the instructions [here](https://stackoverflow.com/c/expensify/questions/13647).

If this regression could have been avoided please consider also proposing a recommendation to the PR checklist so that we can avoid it in the future.

**Comment by MelvinBot** [comment]
⚠️ Looks like this issue was linked to a **Deploy Blocker** [here](https://github.com/Expensify/App/issues/16639)

If you are the assigned CME please investigate whether the linked PR caused a regression and leave a comment with the results.

If a regression has occurred and you are the assigned CM follow the instructions [here](https://stackoverflow.com/c/expensify/questions/13647).

If this regression could have been avoided please consider also proposing a recommendation to the PR checklist so that we can avoid it in the future.

**Comment by stitesExpensify** [comment]
We ended up reverting this PR because of regressions in other unrelated features. The plan moving forward is:
1. We wait for https://github.com/Expensify/App/pull/16356 to be merged
2. We can re-add the changes from the PR (in a new PR)
3. Make sure that all above regressions no longer occur

**Comment by JmillsExpensify** [comment]
Nice, thanks for the breakdown. It looks like #16356 is moving along and should merge not to long from now in any case.

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.91-1 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/14652


If no regressions arise, payment will be issued on 2023-04-05. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
@JmillsExpensify, @stitesExpensify, @perunt Eep! 4 days overdue now. Issues have feelings too...

**Comment by MelvinBot** [comment]
@JmillsExpensify, @stitesExpensify, @perunt Huh... This is 4 days overdue. Who can take care of this?

**Comment by stitesExpensify** [comment]
Not overdue, we are still waiting on https://github.com/Expensify/App/pull/16356

**Comment by JmillsExpensify** [comment]
Same as last week.

**Comment by JmillsExpensify** [comment]
Margelo is still working through the linked PR that is holding this issue.

**Comment by JmillsExpensify** [comment]
Nice one! @perunt make sure to add this payment to the monthly margelo invoice. @stitesExpensify What do you think about regressions? Is there anything we need to test in particular, apart from:

 - Making sure search is there
 - Searching an emoji works
 
 Anything else?

**Comment by perunt** [comment]
@JmillsExpensify, in regards to the payment, typically I include the total amount of hours spent on the task in the invoice. However, if there is a specific requirement to include it separately, please let me know, and I will make the necessary adjustments to the invoice accordingly

Regarding testing, apart from making sure the search is functioning and searching for an emoji works, there are a couple of additional considerations:
* Testing the consistency of the modal's height and ensuring it is not overlapped with the Status Bar.
* Verifying that the modal's dimensions remain unchanged when switching between picking an emoji from the list and searching for an emoji.

**Comment by perunt** [comment]
@JmillsExpensify should I add something else to my invoice? I've heard about issues with bounties but have no clue if this is such a task. I've distinguished them by price title.

**Comment by stitesExpensify** [comment]
Bump @JmillsExpensify. I don't think anything else needs to be added to the invoice right?

**Comment by JmillsExpensify** [comment]
Ah sorry guys! Yes, this should be added to the invoice. As long as you link the GH issue that's great.

**Comment by JmillsExpensify** [comment]
Regarding the testing steps, do we have anything from the PR that I can copy over for QA?

**Comment by stitesExpensify** [comment]
1. Create a new account.
2. Open a new chat conversation.
3. Access the emoji picker by clicking on the emoji icon in the chat input field.
4. Make sure the search bar is visible at the top of the emoji picker.
5. Type "hand" in the search bar and verify that the list of emojis is updated to show relevant hand emojis.
6. Change the skin tone of an emoji using the appropriate option.
7. Verify that the skin tone of the selected emoji is updated.
8. Type the random string "handsmsmsm" in the search bar.
9. Verify that the message "No results found" is displayed.
10. Clear the search bar by deleting all text.
11. Verify that the emoji picker returns to its starter state with all emojis visible and the search bar empty.

**Comment by stitesExpensify** [comment]
@JmillsExpensify I posted some test steps that we can copy, and then I think we can close this

**Comment by JmillsExpensify** [comment]
Forgot to circle back and create the regression issue. Doing that now.

**Comment by JmillsExpensify** [comment]
Done!
